src/gui/qgscolorwidgets.cpp

```diff
@@ -92,6 +92,26 @@ QPixmap QgsColorWidget::createDragIcon( const QColor &color )
   return pixmap;
 }
 
+QgsColorWidget::ComponentUnit QgsColorWidget::componentUnit( ColorComponent component )
+{
+  switch ( component )
+  {
+    case QgsColorWidget::Hue:
+      return ComponentUnit::Degree;
+    case QgsColorWidget::Saturation:
+    case QgsColorWidget::Value:
+    case QgsColorWidget::Alpha:
+    case QgsColorWidget::Cyan:
+    case QgsColorWidget::Magenta:
+    case QgsColorWidget::Yellow:
+    case QgsColorWidget::Black:
+      return ComponentUnit::Percent;
+
+    default:
```

Comment:
Can you drop the "default" and explicitly list all values instead? Then we'd get explicit warnings if additional components are added in future which should be correctly handled here.